trie-db/src/lookup.rs

```diff
@@ -655,4 +668,174 @@ where
 		}
 		Ok(None)
 	}
+
+	/// Look up the merkle value (hash) of the node that is the closest descendant for the provided
+	/// key.
+	///
+	/// When the provided key leads to a node, then the merkle value of that node
+	/// is returned. However, if the key does not lead to a node, then the merkle value
+	/// of the closest descendant is returned. `None` if no such descendant exists.
+	fn lookup_first_descendent_without_cache(
+		mut self,
+		nibble_key: NibbleSlice,
+		full_key: &[u8],
+	) -> Result<Option<TrieHash<L>>, TrieHash<L>, CError<L>> {
```

Comment:
How can this return `None`? If there doesn't exist any branch of the key, we should return the storage root?

Reply:
Forget what I said, I now realized the change to "closest descendant"  :see_no_evil: